memory/training/siamese/trainer.py

```diff
@@ -175,3 +168,26 @@ def _setup(self):
         # build train and val data generators
         self._train_gen, self._val_gen = self._build_data_generators(train_dataset, val_dataset)
 
+
+    def neighbor_prediction(self):
+        predictions = []
+
+        cache_dir = self.cache_dir
+        test_img_dir = self.test_dir
+
+        neighbor_gen = NeighborGenerator(self)
+        neighbor_gen.load_data(cache_dir)
+
+        # form [[img_file, [neighbors]], ...]
+        neighbors = neighbor_gen.batch_predict(test_img_dir)
```

Comment:
Can we make this into a more specific predict function that takes in a list of images and queries those images?